docs/blog/2022-08-12-observing-randomness.md

```diff
@@ -0,0 +1,140 @@
+- What is the current state of the node related to the ceremony.
+
+We also came up with the idea of using the dashboard for ongoing monitoring of the network (not only during ceremonies), for which we added the following for each beacon in each node:
+
+- When was the last randomness generated.
+- Which is the last round generated by the node.
+- What is the time difference between this node and other nodes (including network round-trip times).
+
+The drand code was already instrumented to provide some of the needed metrics via a Prometheus HTTP endpoint, while others needed to be added.
+
+## Inter-node communications
+
+As mentioned above, one of the requirements was to be able to view the state of the whole drand network without requiring node operators to open any additional holes in their firewall, nor to connect to any hosts other than those included in the drand network itself. Therefore, it was necessary to enable the possibility of fetching a remote node’s metrics via the same method used to coordinate randomness generation: that is, through GRPC.
+
+We also wanted to reduce the amount of metrics traffic between nodes, so we went with a pull system where any node can pull any other node’s metrics when requested. For that we exposed an additional HTTP endpoint that includes the peer’s ID to get Prometheus metrics for the associated peer. A request to that endpoint makes a gRPC call to the remote node to get the metrics before returning them to the client. With this we achieved the ability to get metrics from all nodes in the drand network, without any additional unneeded traffic.
+
+The Protocol Labs nodes all run instances of [Telegraf](https://www.influxdata.com/time-series-platform/telegraf/), whose configuration is hard-coded to get the metrics from all nodes in the drand network. This is possible because the nodes in the network are static and only change during ceremonies. Telegraf then pushes the metrics into [InfluxDB](https://www.influxdata.com/). We then use [Grafana](https://grafana.com/) to visualize the data.
+
+Regarding LoE members, the fact that metrics are exposed using the industry-standard Prometheus format and that any node can gather metrics from any other node in the network makes it possible for any of them to create a setup similar to this and gather metrics into their own observability solution.
+
+## Dashboard design
+
+As is usually the case, it took multiple iterations to get to a dashboard design that was useful and also comprehensible. These iterations involved both changing the layout and organization of the presented information, as well as adding metrics that were missed during initial analysis. For example, during ceremonies some of the state changes happened so fast that they were missed by the polling and dashboard refresh intervals, so we added timestamps to be able to visualize whether a state had changed or not. We also reformulated some metrics labels to reduce cardinality.
```

Comment:
```suggestion
As is usually the case, it took multiple iterations to get to a dashboard design that was useful and also comprehensible. These iterations involved both changing the layout and organization of the represented information, as well as adding metrics that were missed during initial analysis. For example, during ceremonies some of the state changes are so fast that they were missed by the polling and dashboard refresh intervals, so we added timestamps to be able to visualize whether a state had changed or not. We also reformulated some metrics labels to reduce cardinality.
```

Reply:
disagree with represented suggestion

Reply:
yeah the information is not represented, it is presented in a dashboard. I do agree with `happened` -> `are`, though.